modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverSetUtils.java

```diff
@@ -188,82 +200,114 @@ private static ValueResolver getParameterValueResolver(String parameterName, Met
-                                       getDefaultValue(type),
-                                       getExpressionSupport(objectType), false, modelProperties, actualAcceptReferences));
-        } catch (MuleException e) {
-          throw new MuleRuntimeException(e);
         }
-      }
 
-      @Override
-      protected void defaultVisit(MetadataType metadataType) {
-        resolverReference.set(getValueResolverFor(parameterName, metadataType, value, getDefaultValue(type),
-                                                  getExpressionSupport(metadataType), false, modelProperties, acceptsReferences));
-      }
+        @Override
+        protected void defaultVisit(MetadataType metadataType) {
+          resolverReference.set(getValueResolverFor(parameterName, metadataType, value, getDefaultValue(type),
+                                                    getExpressionSupport(metadataType), false, modelProperties,
+                                                    acceptsReferences, valueResolverFactory));
+        }
 
-      private ValueResolver getValueResolverFor(String parameterName, MetadataType metadataType, Object value,
-                                                Object defaultValue, ExpressionSupport expressionSupport, boolean required,
-                                                Set<ModelProperty> modelProperties, boolean acceptsReferences) {
-        return valueResolverFactory.of(parameterName, metadataType, value, defaultValue,
-                                       expressionSupport, required, modelProperties, acceptsReferences);
-      }
 
-    });
+
+      });
+    } else {
+      Object convertedValue = convertValueWithExpressionLanguage(value, type, muleContext.getExpressionManager());
+      resolverReference.set(getValueResolverFor(parameterName, type, convertedValue, null, NOT_SUPPORTED, false, modelProperties,
+                                                false, valueResolverFactory));
+    }
     initialiseIfNeeded(resolverReference.get(), muleContext);
     return resolverReference.get();
   }
 
+  private static Object convertValueWithExpressionLanguage(Object value, MetadataType type, ExpressionManager expressionManager) {
+    return expressionManager.evaluate("#[" + PAYLOAD + "]",
+                                      DataType.builder().type(getType(type).orElse(Object.class))
+                                          .mediaType(getFirstValidMimeType(type))
+                                          .build(),
+                                      BindingContext.builder().addBinding(PAYLOAD, TypedValue.of(value)).build())
+        .getValue();
+  }
+
+  private static MediaType getFirstValidMimeType(MetadataType type) {
+    Collection<String> validMimeTypes = type.getMetadataFormat().getValidMimeTypes();
+    String mimeType;
+    if (validMimeTypes.isEmpty()) {
+      mimeType = "application/json";
```

Comment:
why json and not java?

Reply:
also use constants

Reply:
I do agree that the default is java, but when the code gets here, the metadata format is not APPLICATION/JAVA. The list should not be empty, but will change that default to java to reflect the real default.